### PR TITLE
ARROW-2395: [Python] Fix flake8 warnings outside of pyarrow/ directory. Check in CI

### PR DESCRIPTION
**Pull request by wesm**

### CODE REVIEW

File: python/benchmarks/common.py [hunk 1]
```diff
@@ -19,7 +19,6 @@
 import decimal
 from functools import partial
 import itertools
-import os
```
Comment (xhochy) [inline python/benchmarks/common.py]: flake8 fails on line 41 as it's Python3 only. We should either only execute it for Python 3 or make it Python2 compatible.

**Review by xhochy — CHANGES_REQUESTED** [review]

File: python/benchmarks/common.py (hunk 1 above)
Comment (wesm) [reply python/benchmarks/common.py]: fixed

**Comment by wesm** [comment]
+1, will merge on passing build

**Review by xhochy — APPROVED** [review]
+1, LGTM

**Pull request closed without merging**